### PR TITLE
Dnd5e biography support

### PR DESCRIPTION
**Pull request by cyelis1224**
ezgif-1-f6e577b661

Added a checkbox to the Unkenniness Parameters so that DM can choose to include the biography section of the actors sheet in the preamble being sent to the LLM. This makes it where if the actor already has a Biography you can just add some additional instructions in the preamble to explain how the NPC should talk or behave while handling their backstory from their Bio.

### CODE REVIEW

File: src/scripts/chat-message-response.js [hunk 1]
```diff
@@ -22,11 +22,20 @@ async function triggerResponse(actor, request) {
...
         return;
     }
 
+    if (await actor.getFlag("unkenny", "includeBiography") && parameters.biography.trim()) {
+        preamble += `\n\n${parameters.biography}`;
+    }
+
+    console.log("Constructed Preamble:", preamble);
+
+    request = `${preamble}\n\n${request}`;
```
Comment (TheComamba) [inline src/scripts/chat-message-response.js:37]: This way the preamble is included twice:
`generateResponse` calls `collectChatMessages`, which calls `messagesOrganisedForTemplate`, which prepends the preamble.

File: src/scripts/chat-message-response.js [hunk 2]
```diff
@@ -22,11 +22,20 @@ async function triggerResponse(actor, request) {
...
+
```
Comment (TheComamba) [inline src/scripts/chat-message-response.js:38]: Also, `messagesOrganisedForTemplate` currently throws an error if the preamble is falsy (e.g. if it is empty), but with this feature I don't think this should necessarily be an error any more. The logic is more complicated now.
(The function resides in collecting-chat-messages.js, but I think I can only comment files that changed with this PR.)

**Comment by TheComamba** [comment]
Are you willing to add some tests like "bio is included if boolean is set", "included bio and empty preamble is ok", "included empty bio and empty preamble is error", "empty preamble and no bio inclusion is error", or shall I?

**Review by TheComamba — CHANGES_REQUESTED** [review]

**Comment by cyelis1224** [comment]
I can handle it. I'll take care of it tomorrow.

**Comment by cyelis1224** [comment]
So I can split this out if need be, but I just wanted to show I got the biography working like you requested, I just accidentally did it in another branch that I had for reworking the way conversations are handled by the LLM. I'll split it if you don't want to include all this for now but wanted to show you the changes I had to make to get multi-user convo's working and some finer control over the LLM's instructions.

File: src/scripts/chat-message-response.js (hunk 2 above)
Comment (cyelis1224) [reply src/scripts/chat-message-response.js:48]: This should be fixed in this commit but I'll have to split it from the other changes I made for the conversation mechanics.

**Comment by cyelis1224** [comment]
I went ahead and split it, I also removed the checkbox as it felt redundant since we are checking for one the other or both now.

**Comment by cyelis1224** [comment]
Alright, I'll see what I can figure out. 

I saw in the issue thread for the chat window conversation we were having you said you were assuming I was a better front end dev. I highly doubt that, I work in construction and just play with JS code as a hobby. 

I was one of the earlier contributors for the AboveVTT project (A VTT extension that overlays a VTT directly in dndbeyond.com) but I swapped to foundry a while back and recently started playing with making modules for Foundry. Happened across your project and thought it was interesting and was just playing around haha.

**Comment by TheComamba** [comment]
Since you are a contributor now, I suggest that we merge these changes to a branch in the main repository. This will make, well, collaboration easier, as we can both have the same repository cloned locally. If it's allright with you.
I think you should have the permissions to complete this PR yourself, because I changed the target branch to something other than main. However, GitHub will only allow squash merges. If you insist on keeping your history before we go to main I can temporarily disable this rule.
